File: 1.0/src/cz/mg/vulkan/PFNvkGetMemoryHostPointerPropertiesEXT.java
package cz.mg.vulkan;

public class PFNvkGetMemoryHostPointerPropertiesEXT extends VkFunctionPointer {
    public PFNvkGetMemoryHostPointerPropertiesEXT() {
    }

    protected PFNvkGetMemoryHostPointerPropertiesEXT(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkGetMemoryHostPointerPropertiesEXT(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkGetMemoryHostPointerPropertiesEXT(long value) {
        setValue(value);
    }

    public PFNvkGetMemoryHostPointerPropertiesEXT(VkInstance instance) {
        super(instance, new VkString("vkGetMemoryHostPointerPropertiesEXT"));
    }

    public void call(VkDevice device, VkExternalMemoryHandleTypeFlagBits handleType, VkObject pHostPointer, VkMemoryHostPointerPropertiesEXT pMemoryHostPointerProperties, VkResult rval){
        callNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), handleType != null ? handleType.getVkAddress() : VkPointer.getNullAddressNative(), pHostPointer != null ? pHostPointer.getVkAddress() : VkPointer.NULL, pMemoryHostPointerProperties != null ? pMemoryHostPointerProperties.getVkAddress() : VkPointer.NULL, rval != null ? rval.getVkAddress() : VkPointer.getSinkAddressNative());
    }

    protected static native void callNative(long vkaddress, long device, long handleType, long pHostPointer, long pMemoryHostPointerProperties, long rval);


    public int call(VkDevice device, int handleType, VkObject pHostPointer, VkMemoryHostPointerPropertiesEXT pMemoryHostPointerProperties){
        return callSimplifiedNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), handleType, pHostPointer != null ? pHostPointer.getVkAddress() : VkPointer.NULL, pMemoryHostPointerProperties != null ? pMemoryHostPointerProperties.getVkAddress() : VkPointer.NULL);
    }

    protected static native int callSimplifiedNative(long vkaddress, long device, int handleType, long pHostPointer, long pMemoryHostPointerProperties);

}

File: 1.0/src/c/cz/mg/vulkan/PFN_vkGetMemoryHostPointerPropertiesEXT.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkGetMemoryHostPointerPropertiesEXT_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong handleType, jlong pHostPointer, jlong pMemoryHostPointerProperties, jlong rval){
    (void)env;
    (void)jc;
    PFN_vkGetMemoryHostPointerPropertiesEXT f = (PFN_vkGetMemoryHostPointerPropertiesEXT)jniLongToFunctionPointer(address);
    VkResult* rvalAddress = jniLongToPointer(rval);
    *rvalAddress = f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkExternalMemoryHandleTypeFlagBits*)jniLongToPointer(handleType)),
        ((void*)jniLongToPointer(pHostPointer)),
        ((VkMemoryHostPointerPropertiesEXT*)jniLongToPointer(pMemoryHostPointerProperties))
    );
}


jint Java_cz_mg_vulkan_PFNvkGetMemoryHostPointerPropertiesEXT_callSimplifiedNative(JNIEnv* env, jclass jc, jlong address, jlong device, jint handleType, jlong pHostPointer, jlong pMemoryHostPointerProperties){
    (void)env;
    (void)jc;
    PFN_vkGetMemoryHostPointerPropertiesEXT f = (PFN_vkGetMemoryHostPointerPropertiesEXT)jniLongToFunctionPointer(address);
    return f(
        *((VkDevice*)jniLongToPointer(device)),
        (VkExternalMemoryHandleTypeFlagBits)handleType,
        ((void*)jniLongToPointer(pHostPointer)),
        ((VkMemoryHostPointerPropertiesEXT*)jniLongToPointer(pMemoryHostPointerProperties))
    );
}

